fdbserver/CommitProxyServer.actor.cpp

```diff
@@ -1291,13 +1267,21 @@ void determineCommittedTransactions(CommitBatchContext* self) {
 // This first pass through committed transactions deals with "metadata" effects (modifications of txnStateStore, changes
 // to storage servers' responsibilities)
 ACTOR Future<Void> applyMetadataToCommittedTransactions(CommitBatchContext* self) {
-	auto pProxyCommitData = self->pProxyCommitData;
-	const auto& trs = self->trs;
+	state ProxyCommitData* const pProxyCommitData = self->pProxyCommitData;
+	state std::unordered_map<EncryptCipherDomainId, Reference<BlobCipherKey>> defaultCipherKeys;
+	if (pProxyCommitData->encryptMode.isEncryptionEnabled()) {
+		std::unordered_map<EncryptCipherDomainId, Reference<BlobCipherKey>> cks = wait(
+		    getLatestEncryptCipherKeys(pProxyCommitData->db, ENCRYPT_CIPHER_DETAULT_DOMAINS, BlobCipherMetrics::TLOG));
```

Comment:
The code here would result it fetching EKs for: systemKeySpace , HeaderDomain and DefaultDomain; it might be an overkill, especially if the cluster is configured with domain_aware semantics, default_domain keys may lead to RPC to external KMS which can be avoided.
Also, if KMS supports encryption-key rotations, then then accesses are costly due to new EK generated by external KMS (which later needs to be cleaned up as well).
Same logic applies to HeaderDomain keys as well (encryption authentication if disabled..)

Better approach would be do the following:
1. Pull system-key space always (keeps the cache warm, useful operation)
2. Pull `default_domain` if mode == `cluster_aware`
3. Pull `header_domain` if `encryption_authentication` is enabled.

Thoughts?